File: http/http-server-annotation-processor/src/test/java/ru/tinkoff/kora/http/server/annotation/processor/controller/ParentTestController.java
package ru.tinkoff.kora.http.server.annotation.processor.controller;

import ru.tinkoff.kora.http.common.HttpMethod;
import ru.tinkoff.kora.http.common.annotation.HttpRoute;

public class ParentTestController<T> {
    @HttpRoute(method = HttpMethod.GET, path = "/parent")
    public T someMethod() {
        return null;
    }

    @HttpRoute(method = HttpMethod.POST, path = "/parent-param")
    public void someMethodWithParam(T param) {
    }


}

File: http/http-server-symbol-processor/src/test/kotlin/ru/tinkoff/kora/http/server/symbol/processor/controllers/ParentTestController.kt
package ru.tinkoff.kora.http.server.symbol.processor.controllers

import ru.tinkoff.kora.http.common.HttpMethod
import ru.tinkoff.kora.http.common.annotation.HttpRoute

open class ParentTestController<T> {
    @HttpRoute(method = HttpMethod.GET, path = "/parent")
    open fun someMethod(): T? {
        return null
    }

    @HttpRoute(method = HttpMethod.POST, path = "/parent-param")
    open fun someMethodWithParam(param: T) {
    }
}
